### PR TITLE
[BD-6] OEP-18 Compliance

### PR DESCRIPTION
**Pull request by ericfab179**
Update setup.py install_requirements definition.
Create requirements folder.
create pip_tools, base, dev, test, tox and travis requirements files.
Remove old requirement files.
Add upgrade command to makefile.
Include requirements files generated using upgrade command.
Update openedx.yml OEPs list.

**Reviewers**
- [ ] @awais786 
- [ ] @morenol

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @ericfab179! I've created [OSPR-4554](https://openedx.atlassian.net/browse/OSPR-4554) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: requirements/travis.in [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Requirements for running tests in Travis
+-c constraints.txt
+
+-r test.txt
```
Comment (morenol) [inline requirements/travis.in:4]: You can remove this, travis only needs to install tox

Comment (morenol) [reply requirements/travis.in:4]: And codecov, you should add codecov to this file

**Comment by edx-webhook** [comment]
@ericfab179 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/staff_graded-xblock&entry.867055334=https://github.com/edx/staff_graded-xblock/pull/14&entry.1484655318=https://github.com/ericfab179&entry.752974735=2020-05-18+23:57&entry.1917517419=2020-05-20+09:35&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**